python/lsst/ctrl/mpexec/job_report.py

```diff
@@ -0,0 +1,187 @@
+# This file is part of ctrl_mpexec.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ["ExecutionStatus", "JobReport", "QuantumReport"]
+
+import enum
+import shlex
+import sys
+from typing import Dict, List, Optional
+
+from lsst.daf.butler import DataCoordinate, DataId, DataIdValue
+from pydantic import BaseModel
+
+
+def _cmdLine() -> str:
+    """Return command line for current process."""
+    return shlex.join(sys.argv)
+
+
+def _serializeDataId(dataId: DataId) -> Dict[str, DataIdValue]:
+    if isinstance(dataId, DataCoordinate):
+        return dataId.byName()
+    else:
+        return dataId  # type: ignore
+
+
+class ExecutionStatus(enum.Enum):
+    """Possible values for job execution status.
+
+    Status `FAILURE` is set if one or more tasks failed. Status `TIMEOUT` is
+    set if there are no failures but one or more tasks timed out. Timeouts can
+    only be detected in multi-process mode, child task is killed on timeout
+    and usually should have non-zero exit code.
+    """
+
+    SUCCESS = "success"
+    FAILURE = "failure"
+    TIMEOUT = "timeout"
+
+
+class ExceptionInfo(BaseModel):
+    """Information about exception."""
+
+    className: str
+    """Name of the exception class if exception was raised."""
+
+    message: str
+    """Exception message for in-process quantum execution, None if
+    quantum was executed in sub-process.
+    """
+
+    def __init__(self, exception: Exception):
+        """Construct instance from an exception."""
+        super().__init__(className=exception.__class__.__name__, message=str(exception))
```

Comment:
In theory we can use `lsst.utils.introspection.get_full_type_name()` -- depends on whether you want custom exceptions to include the full type name or just the short part.